src/framework/audio/iaudiosource.h

```diff
@@ -39,13 +39,7 @@ class IAudioSource
     virtual async::Channel<unsigned int> streamsCountChanged() const = 0;
 
     //! move buffer forward for sampleCount samples
-    virtual void forward(unsigned int sampleCount) = 0;
-
-    //! const access to the source's output buffer
-    virtual const float* data() const = 0;
-
-    //! set the mix step in samples
-    virtual void setBufferSize(unsigned int samples) = 0;
+    virtual void forward(float* buffer, unsigned int sampleCount) = 0;
```

Comment:
I think better rename it to just `process`

Reply:
I was thinking about it, but wasn't sure about the "process" since we also have IAudioProcessor::process
Anyway, let's discuss it directly tomorrow